### PR TITLE
test(ks,youth): improve test coverage for youth UI

### PR DESCRIPTION
**Pull request by nikomakela**
YJDH-772.

Add tests for server backend, i18n loaders,
components, hooks, type guards, and helper edge cases to resolve coverage gaps.

## Summary by CodeRabbit

* **Tests**
  * Added test coverage for form submission behavior, error handling, and validation workflows.
  * Expanded test coverage for form error summary rendering, field selection components, and target group functionality.
  * Introduced comprehensive test suites for application creation operations and utility functions.

### CODE REVIEW

File: frontend/kesaseteli/youth/src/__tests__/index.validation.test.tsx [hunk 1]
```diff
@@ -250,5 +252,28 @@ describe('frontend/kesaseteli/youth/src/pages/index.tsx', () => {
         );
       });
     });
+
+    describe('getIndexPageApi helper edge cases', () => {
+      it('exercises key === selectedSchool in textInputIsValid', async () => {
```
Comment (karisal-anders) [inline frontend/kesaseteli/youth/src/__tests__/index.validation.test.tsx]: What is the test case when `=== selectedSchool` happens? Now the test case is about `=== selectedSchool` but it'd be clearer IMHO to state what test case is being tested on a more general level, not just the code logic when it happens, but when what the actual case is. Not a huge deal, but could bring clarity to this IMHO.

File: frontend/kesaseteli/youth/src/components/youth-form/__tests__/SubmitErrorSummary.test.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    renderWithForm(<SubmitErrorSummary error={error} />);
+    expect(
+      screen.getByText('common:youthApplication.checkNotification.recheck')
+    ).toBeInTheDocument();
+  });
+
+  it('renders validation error notification with error fields', () => {
+    const error: SubmitError = {
+      type: 'validation_error',
+      errorFields: ['first_name' as any, 'last_name' as any],
```
Comment (karisal-anders) [inline frontend/kesaseteli/youth/src/components/youth-form/__tests__/SubmitErrorSummary.test.tsx]: These work fine without the `as any`s:
```suggestion
      errorFields: ['first_name', 'last_name'],
```

File: frontend/kesaseteli/youth/src/components/youth-form/__tests__/SubmitErrorSummary.test.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    // Silence error log of assertUnreachable
+    const originalConsoleError = console.error;
+    console.error = jest.fn();
+
+    expect(() => {
+      renderWithForm(<SubmitErrorSummary error={error} />);
+    }).toThrow('Unknown submit error type: unknown_type');
+
+    console.error = originalConsoleError;
```
Comment (karisal-anders) [inline frontend/kesaseteli/youth/src/components/youth-form/__tests__/SubmitErrorSummary.test.tsx]: One way to make the console.error overriding more robust (i.e. not break if something happens in between the overriding and restore) would be:
```suggestion
    try {
      expect(() => {
        renderWithForm(<SubmitErrorSummary error={error}/>);
      }).toThrow('Unknown submit error type: unknown_type');
    } finally {
      console.error = originalConsoleError;
    }
```

Probably others too with spyOn etc.

File: frontend/kesaseteli/youth/src/hooks/__tests__/useHandleYouthApplicationSubmit.test.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+    act(() => {
+      result.current.handleErrorResponse(creationError);
+    });
+
+    expect(mockPush).toHaveBeenCalledWith('/fi/already_assigned');
+  });
+
+  it('handleErrorResponse sets error summary on please_recheck_data', () => {
+    (isYouthApplicationCreationError as unknown as jest.Mock).mockReturnValue(
```
Comment (karisal-anders) [inline frontend/kesaseteli/youth/src/hooks/__tests__/useHandleYouthApplicationSubmit.test.tsx]: I tried this and "yarn typecheck" passed and the tests passed with this:
```suggestion
    jest.mocked(isYouthApplicationCreationError).mockReturnValue(
```

Use of `jest.mocked` gets rid of all the `as unknown as` typecasts in this whole file for both `isYouthApplicationValidationError` and `isYouthApplicationCreationError`.

File: frontend/kesaseteli/youth/src/hooks/__tests__/useHandleYouthApplicationSubmit.test.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    });
+
+    expect(result.current.submitError).toEqual({
+      type: 'please_recheck_data',
+      errorFields: [],
+    });
+    expect(mockReset).toHaveBeenCalled();
+  });
+
+  it('handleErrorResponse asserts unreachable for unknown code', () => {
```
Comment (karisal-anders) [inline frontend/kesaseteli/youth/src/hooks/__tests__/useHandleYouthApplicationSubmit.test.tsx:104]: A small detail, which you can consider whether you see this as an improvement:
```suggestion
  it('handleErrorResponse throws an error for unknown code', () => {
```

I think this would make the test case a bit more transparent.

**Review by karisal-anders — APPROVED** [review]
Left some comments, please take a look at them.

The backend-server.test.ts & backend-server.ts looks like an integration test. Is that useful here?

File: frontend/kesaseteli/youth/src/lib/backend-server.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import SummerVoucherConfiguration from 'kesaseteli-shared/types/summer-voucher-configuration';
+
+export async function getConfiguration(): Promise<
```
Comment (nikomakela) [inline frontend/kesaseteli/youth/src/lib/backend-server.ts]: I should get rid of this. Accidentally pushed scratch stuff.

File: frontend/kesaseteli/youth/src/lib/i18n-server.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable security/detect-non-literal-fs-filename */
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Server-side translation loader for App Router Server Components.
+ * Bypasses React Context to allow static generation and localized metadata.
+ */
+export async function getCommonTranslations(locale: string): Promise<any> {
```
Comment (nikomakela) [inline frontend/kesaseteli/youth/src/lib/i18n-server.ts]: I should get rid of this. Accidentally pushed scratch stuff.
